index.js

```diff
@@ -51,7 +53,10 @@ module.exports = function (str, opts) {
 	}
 
 	var ret = lines.map(function (line, lineIndex) {
-		var ret = {};
+		var ret = headers.reduce(function (ret, header) {
+			ret[header] = '';
+			return ret;
+		}, {});
```

Comment:
Lines that are shorter than the header might miss the last few columns, so this goes through and sets each column to a default `''` (which makes sense since values will always be strings).